src/UglyToad.PdfPig/DocumentLayoutAnalysis/DocstrumBB.cs

```diff
@@ -0,0 +1,212 @@
+    /// clustering of connected components extracted from the document. 
+    /// This implementation leverages bounding boxes and does not exactly replicates the original algorithm.
+    /// <para>See 'The document spectrum for page layout analysis.' by L. O’Gorman.</para>
+    /// </summary>
+    public class DocstrumBB : IPageSegmenter
+    {
+        /// <summary>
+        /// Create an instance of Docstrum for bounding boxes page segmenter, <see cref="DocstrumBB"/>.
+        /// </summary>
+        public static DocstrumBB Instance { get; } = new DocstrumBB();
+
+        /// <summary>
+        /// Get the blocks.
+        /// <para>Uses wlAngleLB = -30, wlAngleUB = 30, blAngleLB = -135, blAngleUB = -45, blMulti = 1.3.</para>
+        /// </summary>
+        /// <param name="pageWords"></param>
+        /// <returns></returns>
+        public IReadOnlyList<TextBlock> GetBlocks(IEnumerable<Word> pageWords)
+        {
+            return GetBlocks(pageWords, -30, 30, -135, -45, 1.3);
+        }
+
+        /// <summary>
+        /// Get the blocks. See original paper for more information.
+        /// </summary>
+        /// <param name="pageWords"></param>
+        /// <param name="wlAngleLB">Within-line lower bound angle.</param>
+        /// <param name="wlAngleUB">Within-line upper bound angle.</param>
+        /// <param name="blAngleLB">Between-line lower bound angle.</param>
+        /// <param name="blAngleUB">Between-line upper bound angle.</param>
+        /// <param name="blMultiplier">Multiplier that gives the maximum perpendicular distance between 
+        /// text lines for blocking. Maximum distance will be this number times the between-line 
+        /// distance found by the analysis.</param>
+        /// <returns></returns>
+        public IReadOnlyList<TextBlock> GetBlocks(IEnumerable<Word> pageWords, double wlAngleLB, double wlAngleUB,
+            double blAngleLB, double blAngleUB, double blMultiplier)
+        {
+            var pageWordsArr = pageWords.Where(w => !string.IsNullOrWhiteSpace(w.Text)).ToArray(); // remove white spaces
+
+            var withinLineDistList = new ConcurrentBag<double[]>();
+            var betweenLineDistList = new ConcurrentBag<double[]>();
+
+            // 1. Estimate in line and between line spacing
+            Parallel.For(0, pageWordsArr.Length, i =>
+            {
+                var word = pageWordsArr[i];
+
+                // Within-line distance
+                var pointWL = GetNearestPointData(pageWordsArr, word,
+                    bb => bb.BottomRight, bb => bb.BottomRight,
+                    bb => bb.BottomLeft, bb => bb.BottomLeft,
+                    wlAngleLB, wlAngleUB, Distances.Horizontal);
+                if (pointWL != null) withinLineDistList.Add(pointWL);
+
+                // Between-line distance
+                var pointBL = GetNearestPointData(pageWordsArr, word,
+                    bb => bb.BottomLeft, bb => bb.Centroid,
```

Comment:
I wonder if this calculation of between line distance can be thrown off by letters with descenders/ascenders. I'm guessing because I'm not clever enough to understand the algorithm but would `letter.Location` work better since it represents the shared baseline of letters on a line: https://github.com/UglyToad/PdfPig/wiki/Letters

Though I guess that doesn't work for vertical/rotated text?

Reply:
Oh this is words not letters, ignore me.

Reply:
Hi, event if it uses `Word` and not `Letter`, I'll have a look as descenders/ascenders might have an impact. Also, good point on Rotated and Vertical text, but needs more dev